docs/codeql/codeql-for-visual-studio-code/running-codeql-queries-at-scale-with-mrva.rst

```diff
@@ -0,0 +1,184 @@
+
+TODO: Check on "internal" repositories.
+
+.. pull-quote::
+
+    Note
+
+    You can update your settings to use a different controller repository when you want to run variant analysis on a different group of repositories. For example, if you have finished testing the query on open source code and now want to test it on your private code. However, you must wait until any previous analysis is complete before you change the controller repository.
+
+TODO: check that the guess in the note above is accurate.
+
+Setting up variant analysis
+---------------------------
+
+You can configure the CodeQL extension to run variant analysis by defining a controller repository.
+
+.. image:: ../images/codeql-for-visual-studio-code/controller-repository.png
+    :width: 350
+    :alt: Screenshot of the CodeQL extension in Visual Studio Code. The "Variant Analysis Repositories" section is expanded and the "Set up controller repository" button is highlighted with a dark orange outline.
+
+#. In Visual Studio Code, click **QL** in the left sidebar to display the CodeQL extension.
+
+#. Expand **Variant Analysis Repositories** and click **Set up controller repository** to display a field for the controller repository.
+
+#. Type the owner and name of the repository on GitHub.com that you want to use as your controller repository and press the **Enter** key.
+
+#. If you are prompted to authenticate with GitHub, follow the instructions and sign into your pesonal or organization account. When you have finished following the process, a prompt from GitHub Authentication may ask for permission to open a URI in Visual Studio Code, click **Open**.
+
+The name of the controller repository is saved in your settings for the CodeQL extension. For information on how to edit the controller repository, see ":ref:`Customizing settings <customizing-settings>`."
+
+Running a query at scale using variant analysis
+-----------------------------------------------
+
+#. Expand the **Variant Analysis Repositories** section, to show the default lists of the top 10, top 100, and top 1000 public repositories on GitHub.com. These are ranked by considering various metrics such as number of stars, number of watchers, number of forks etc.
+
+#. Select which GitHub repository or repositories you want to run your query against.
+
+   - Select **Top X repositories**.
+   - Click **Add new database**, the **+** icon, to select either a single repository, or all repositories in an organization.
+   - Click **LIST-NAME** to select a custom list that you have created previously.
+   - Click **Add new list**, the folder icon, to create a custom list (see later in this article for details).
```

Comment:
I hope this will be obvious in the UI.

```suggestion
```

Reply:
I would tend towards keeping this. Unless I've misunderstood, "Add new database" seems to have two uses:

1. Add a repository or organization to the panel, as-is.
2. Add a repository to the currently selected custom list.